packages/provider/src/RelayClient.ts

```diff
@@ -556,6 +561,16 @@ export class RelayClient {
       this.logger.debug(`Reading default client config for chainId ${chainId.toString()}`)
       configFromServer = await this._resolveConfigurationFromServer(chainId, defaultGsnConfig.clientDefaultConfigUrl)
     }
+    if (config.verifierServerApiKey != null && config.verifierServerApiKey.length !== 0) {
```

Comment:
Do we have to have two different checks for the server api key here, as well as nested checks? maybe clean it up a bit, move that part to a different function, and early return instead of nested checks?